Cython/Compiler/Nodes.py

```diff
@@ -1596,7 +1596,7 @@ def analyse_declarations(self, env, dest_scope=None):
                     self.entry.create_wrapper = True
             else:
                 if self.overridable:
-                    error(self.pos, "Variables cannot be declared with 'cpdef'. Use 'cdef' instead.")
+                    error(self.pos, "Variables cannot be declared with 'cpdef'. Use 'cdef' or 'cdef public' instead.")
```

Comment:
I wouldn't make this change.

People were using `cpdef` with the belief that it would also expose the variable through a Python interface. `cdef public` also doesn't do that bit it sounds like it might.  So I think this is more likely to mislead than help.

I'd also consider `cdef public` to be a subset of `cdef` so it doesn't need specifically calling out.

Reply:
Ah you're partly right - sorry.

It only does that in classes which makes it good advice for class attributes but not for module-scoped variables. (Admittedly it's unhelpful that `public` means a different thing there, but that unfortunately historical).